v2/router.go

```diff
@@ -48,9 +48,38 @@ type Procedure struct {
 	Signature string
 }
 
+// EncodingProcedure specifies a single encoding-level handler and is mapped to a TransportProcedure by the router.
+type EncodingProcedure struct {
+	// Name of the procedure.
+	Name string
+
+	// Service or empty to use the default service name.
+	Service string
+
+	// HandlerSpec specifying which handler and rpc type.
+	HandlerSpec EncodingHandlerSpec
+
+	// Encoding of the handler.
+	// (if present).
+	Encoding Encoding
```

Comment:
What's the behavior of an `EncodingProcedure` if this isn't present?

Reply:
I don't believe an `EncodingProcedure` necessarily needs to exist in the eventual procedure/middleware stack - especially if there's no encoding.

Reply:
I think what @amckinney is asking is: 
_If_ we have an `EncodingProcedure`, doesn't this mean we **must** have an encoding present -- and therefore this field **should** always be able to have a value?

Reply:
Oh, yeah - that sounds true.